File: UI/Client/src/app/Models/Deal.ts
import { Asset } from "./Asset";

export class Deal{
    id: number;
    productName: string;
    shortDesc: string;
    longDesc: string;
    categoryId: number;
    subcategoryId: number;
    uom: string;
    quantity:number;
    partNumber: string;
    priceBuyNow:number;
    canBuyNow: boolean;
    isChecked: boolean;
    userId: number;
    assets : Asset[];

    constructor(
        _id: number,
        _productName: string,
        _shortDesc: string,
        _longDesc: string,
        _categoryId: number,
        _subcategoryId: number,
        _uom: string,
        _quantity: number,
        _partNumber: string,
        _priceBuyNow:number,
        _canBuyNow: boolean,
        _isChecked: boolean,
        _userId: number,
        _assets: Asset[]
        )        {
        this.id = _id;
        this.productName = _productName;
        this.shortDesc = _shortDesc;
        this.longDesc = _longDesc;
        this.categoryId = _categoryId;
        this.subcategoryId = _subcategoryId;
        this.uom = _uom;
        this.quantity = _quantity;
        this.partNumber = _partNumber;
        this.priceBuyNow = _priceBuyNow;
        this.canBuyNow = _canBuyNow;
        this.isChecked = _isChecked;
        this.userId = _userId;
        this.assets = _assets;
    }
}

File: FeedAPI/FeedAPI/Common/EntityFramework/Models/Deal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Common.EntityFramework.Models
{
    [Table("deals")]
    public class Deal
    {
        //[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public int Id { get; set; }

        [Column("productname")]
        public string? ProductName { get; set; }

        [Column("shortdesc")]
        public string? ShortDesc { get; set; }

        [Column("longdesc")]
        public string? LongDesc { get; set; }

        [Column("statusid")]
        public int? StatusId { get; set; }

        [Column("categoryid")]
        public int? CategoryId { get; set; }

        [Column("uom")]
        public string? UoM { get; set; }

        [Column("quantity")]
        public double? Quantity { get; set; }

        [Column("partnumber")]
        public string? PartNumber { get; set; }

        [Column("ischecked")]
        public bool? IsChecked { get; set; }

        [Column("canbuynow")]
        public bool? CanBuyNow { get; set; }

        [Column("pricebuynow")]
        public double? PriceBuyNow { get; set; }

        [Column("userid")]
        public int? UserId { get; set; }

        [NotMapped]
        public string? UserFullName { get; set; }

        [Column("startbet")]
        public double? StartBet { get; set; }

        [Column("starttime")]
        public DateTime? StartTime { get; set; }

        [Column("duration")]
        public int? Duration { get; set; }

        [NotMapped]
        public List<Bet> Bets { get; set; }

        [NotMapped]
        public List<Asset> Assets { get; set; }

        [NotMapped]
        public List<WatchDeal> WatchDeals { get; set; }

        [NotMapped]
        public Auction Auction { get; set; }
    }
}
